File: app/src/main/java/com/example/testapplication/kotlintest/Reflection.java
package com.example.testapplication.kotlintest;

import java.util.LinkedList;
import java.util.List;
import java.util.Objects;

public class Reflection {
    List<String> list=new LinkedList<String>();
    List<Object> listO=new LinkedList<Object>();
    List<Object> list1=new LinkedList< >();
    public void add(){
        listO.addAll(list);
        listO=list1;
    }
    public static void main(String[] args){

    }
}

File: app/src/main/java/com/example/testapplication/kotlintest/reflection.kt
package com.example.testapplication.kotlintest

class reflection {
}


fun main(){
    var list= ArrayList<Student>()
    var listA= ArrayList<Man>()
    listA.addAll(list)

//    listA=list
}
//fun getData  (data:out T){
//
//}
open class Man{}
open class Student: Man(){}
open class Teacher: Man(){}
class XiaoMing : Student(){}
class XiaoLi : Student(){}
class LaoWang : Teacher(){}